src/domain/bitcoin/onchain/index.types.d.ts

```diff
@@ -111,3 +111,17 @@ interface IOnChainService {
     targetConfirmations,
   }: PayToAddressArgs): Promise<OnChainTxHash | OnChainServiceError>
 }
+
+interface OnChainEvent {
+  payload: string
+  sequence: string
+  recorded_at: number
+}
+
+type OnChainEventHandler = (event: OnChainEvent) => true | ApplicationError
```

Comment:
ApplicationError is not part of domain layer

Reply:
Think of the event handler as a similar to the graphql layer. It calls into the app.